src/gt4py/stencil_builder.py

```diff
@@ -215,6 +219,7 @@ def with_externals(self: "StencilBuilder", externals: Dict[str, Any]) -> "Stenci
         """
         self._build_data = {}
         self._externals = externals
+        self.with_caching(self.caching.name)
```

Comment:
The `__init__` already set the caching strategy. Is this needed because of a quirk?

Reply:
After changing externals, we need to avoid using a previous cache built around previous externals, so we need to reset it.